src/test/java/org/kitteh/irc/client/library/util/MaskTest.java

```diff
@@ -0,0 +1,114 @@
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * It's a mask test.
+ */
+public class MaskTest {
+    /**
+     * Tests a full string.
+     */
+    @Test
+    public void testFullString() {
+        final Mask mask = Mask.fromString("mbaxter!~mbax@kitten.institute");
+        Assert.assertTrue(mask.getNick().isPresent());
+        Assert.assertEquals("mbaxter", mask.getNick().get());
+        Assert.assertTrue(mask.getUser().isPresent());
+        Assert.assertEquals("~mbax", mask.getUser().get());
+        Assert.assertTrue(mask.getHost().isPresent());
+        Assert.assertEquals("kitten.institute", mask.getHost().get());
+        Assert.assertEquals("mbaxter!~mbax@kitten.institute", mask.asString());
+    }
+
+    /**
+     * Tests a nick.
+     */
+    @Test
+    public void testNick() {
+        final Mask mask = Mask.fromNick("mbaxter");
+        Assert.assertTrue(mask.getNick().isPresent());
+        Assert.assertEquals("mbaxter", mask.getNick().get());
+        Assert.assertFalse(mask.getUser().isPresent());
+        Assert.assertFalse(mask.getHost().isPresent());
+        Assert.assertEquals("mbaxter!*@*", mask.asString());
+    }
+
+    /**
+     * Tests a user string.
+     */
+    @Test
+    public void testUser() {
+        final Mask mask = Mask.fromUser("~mbax");
+        Assert.assertFalse(mask.getNick().isPresent());
+        Assert.assertTrue(mask.getUser().isPresent());
+        Assert.assertEquals("~mbax", mask.getUser().get());
+        Assert.assertFalse(mask.getHost().isPresent());
+        Assert.assertEquals("*!~mbax@*", mask.asString());
+    }
+
+    /**
+     * Tests a host.
+     */
+    @Test
+    public void testHost() {
+        final Mask mask = Mask.fromHost("kitten.institute");
+        Assert.assertFalse(mask.getNick().isPresent());
+        Assert.assertFalse(mask.getUser().isPresent());
+        Assert.assertTrue(mask.getHost().isPresent());
+        Assert.assertEquals("kitten.institute", mask.getHost().get());
+        Assert.assertEquals("*!*@kitten.institute", mask.asString());
+    }
+
+    /**
+     * Tests that a user passes a predicate test.
+     */
+    @Test
+    public void testPredicateTest() {
+        final Mask mask = Mask.fromNick("kashike");
+        final User kashike = new TestUser("kashike", "kashike", "is.a.miserable.ninja", null);
+        final User lol768 = new TestUser("lol768", "lol768", "catastrophe.esper.net", null);
+        Assert.assertTrue(mask.test(kashike));
+        Assert.assertFalse(mask.test(lol768));
+    }
+
+    /**
+     * Tests matches against a channel.
+     */
+    @Test
+    public void testMatches() {
+        final Channel channel = Mockito.mock(Channel.class);
+        final User mbaxter = new TestUser("mbaxter", "~mbax", "kitten.institute", null);
+        final User kashike = new TestUser("kashike", "kashike", "is.a.miserable.ninja", null);
+        final User lol768 = new TestUser("lol768", "lol768", "catastrophe.esper.net", null);
```

Comment:
ideally use a host different than *.esper.net, I'm fine with using lynvie.com for mine or just somethinf spoofed.